### DISCUSSION TITLE
copy/paste claim permissions and flags

### DISCUSSION THREAD

**Discussion by Oranamous** [GriefPrevention/GriefPrevention#2091] [Ideas]
(I am both a player and a developer, so I'm looking for a solution that would be both helpful for players and relatively easy to implement without having to tear the whole plugin apart and put it back together)

**Put simply:**
A way to copy a set of permissions and flags from one claim, and paste it to other owned claims.

**Why it's needed:**
This would be helpful in situations where a group has all claims owned by one person (for the sake of organization in my case), and that one person always has to manually /trust new players in every single claim. Due to the rectangular shape of claims, since our area is oddly shaped and works around others' claims, this results in sometimes up to 30 commands required each time a new player joins the group.

**How it could be implemented**:
/copyperms and /copyflags would save the current permissions and flags respectively to the player's clipboard, this clipboard wouldn't need to be persistent on server restart I feel, or even on relog, but whatever the default behavior of the data is can stay I guess. /pasteperms and /pasteflags could then apply the copied flags to the current claim, but since this is similar to a reset it might be a good idea to add a confirmation before applying changes.

**Other possible solutions**:
 - Allow players to create permission and/or flag templates (which would be persistent for that player), and allow a claim to either A: use normal behavior, or B: be linked to a template. this would probably require a rewrite to claim data storage but I'm not sure exactly how it's stored currently so maybe not.
 - Group claims, not an original idea but figured I'd mention it here. Would almost definitely require a lot more effort to add (unless you've got it hidden up your sleeve already), not much else to say.
 - Allow merging multiple connected claims into 1, probably the most problematic to implement here, probably the most streamlined option but could cause issues with claims on preexisting servers unless handled well.

**Comment by RoboMWM** [comment]
Here was a simple solution I implemented some years ago: https://github.com/TechFortress/EasyTrust

but ya, if you have any suggestions to API to make it easier to make an addon for this, lemme know.

**Comment by Oranamous** [comment]
I've now made a very simple plugin to copy and paste permissions, haven't messed with flags yet.
https://github.com/Oranamous/GriefPreventionClipboard
Lmk if you have any suggestions, very happy with how this turned out.

**Reply by RoboMWM** [reply]
Cool, mind making a post in the addons category in discussions too? Currently this is the easiest place I can think of for people to post their addons and get visibility https://github.com/TechFortress/GriefPrevention/discussions/categories/addons
